test/Utilities/matrix_of_constraints.jl

```diff
@@ -526,6 +526,65 @@ function test_modif()
     @test_throws err MOI.add_constraint(model, func, set)
 end
 
+MOIU.@struct_of_constraints_by_set_types(
+    ZerosOrNot,
+    MOI.Zeros,
+    Union{MOI.Nonnegatives,MOI.Nonpositives},
```

Comment:
Can you clarify why this is needed in more detail?!? Why not just two separate fields.

Reply:
This allows to have one sparse matrix with the two instead of two separate sparse matrices. ECOS needs one sparse matrix with the three non-Zeros cones

Reply:
Sure, but you could concatenate the sparse matrices? It seems weird that we go to all this effort and then allow `Union`

Reply:
I'd argue that this PR is simpler than implementing concatenation in the solver wrapper though. DiffOpt also needs to split the matrix onto equality and non-equality. This is quite common so having this feature is not just for ECOS